appGeo/src/app/game/game-rules.component.ts

```diff
@@ -0,0 +1,202 @@
+import { Component } from '@angular/core';
+
+export const SUCCESS = 10
+export const FAILURE = -10
+
+const MIN_MIN_PLAYERS = 5
+const MIN_FRACTION_KILLERS = 0.2
+
+//all constants in minutes
+const MIN_GAME_LENGTH = 1440
+const DEF_GAME_LENGTH = 10080
+
+//safe length default is 1/3 of game length. Minimum is 1/2 of game length
+
+const DEF_CYCLE_LEN = 1440
+
+//cannot be longer than safe length
+const MIN_VOTE_LENGTH = 60
+
+//minutes after game begins
+const DEF_VOTE_TIME = 960
+
+const MIN_MAX_KILL = 1
+const DEF_MAX_GLOBAL_KILL = Infinity
```

Comment:
Could you explain a bit more what you mean by this DEF_MAX_GLOBAL_KILL? I am a little bit confused about this, especially at value of Infinity. From what I understand this is supposed to represent the maximum amount of people that the killers as a whole can kill in one day. Let's say there are two killers. Game logic would say that this value would just be 2 right? And the Killers have an internal field that tells them if they have reached their daily max quota (in which case they can't kill anyone else).

Reply:
This is the cumulative amount of kills that killers are allowed to make. Say maxSoloKills = 3 and maxGlobalKills = 4. So an individual killer can kill 3 people a day but together they can only kill 4. So if killer A kills 3 people, then killer B can only kill 1 person before they are all blocked off.

This is to allow for the option to limit the rate of people killed a day without having to do it by limiting the kills through distribution. It's set to Infinity as a default as normally this doesn't need to be set.

Reply:
Ahhhhh I see I see. I hadn't thought about that, but that's fair enough. Should this be something that the Player Class gets then? Because current implementation of Killer Subclass is that there is a check if amount of people killed today is less than Maximum amount of People that they can kill in a day. It doesn't take into consideration global Max Kills at all.